source/connection-string/connection-string-spec.rst

```diff
@@ -180,9 +180,9 @@ The values in connection options MUST be URL decoded by the parser. The values c
 - Strings:
     The value
 - Integer:
-    The value parsed as a integer
+  The value parsed as a integer. If the value is the empty string, then a WARN level logging message must be issued and the key is ignored.
 - Boolean:
-    "true" and "false" strings MUST be supported.
+  "true" and "false" strings MUST be supported. If the value is the empty string, then a WARN level logging message must be issued and the key is ignored.
```

Comment:
I don't recall if this behavior being discussed in Slack, but I'm not a fan of logging a warning and ignoring the key, as this behavior differs significantly from a string option, where an empty string would be captured.

In this case, I'd much rather we raise an error and actually interrupt client construction. From the perspective of C and PHP, logging warnings are not a reliable way to inform our users, as they actually have to opt in to collecting log output.

Alternatively, I'd be OK with logging a warning and interpreting the value as the logical "empty" value, which would be zero or false for integers and booleans, respectively. In that case, we at least assign the option and ensure it will be picked up by some other driver component down the line. Ideally, it might even trigger a validation error and loudly inform the user.

My reasoning with preferring a conversion to zero/false over ignoring entirely is motivated by thinking that the user has already indicated they intend to _use_ this option and is more likely to just have forgotten to specify a value. If their intention was to not use the option at all I think they'd have been more likely to omit it entirely.

Reply:
It is my impression that the warn-and-continue behavior was largely a part of the URI options specification, in the added spec tests (for example the numerous warn-and-continue tests in https://github.com/mongodb/specifications/blob/master/source/uri-options/tests/connection-options.yml). I am not seeing existing language in either the URI spec or the connection string spec stating how to handle unrecognized or invalid or missing values for URI options in general, however looking at the connection string spec I believe this test is prior art that ignores unrecognized option value:

https://github.com/mongodb/specifications/blob/master/source/connection-string/tests/valid-warnings.yml#L15

I generally support failing on invalid input (which includes missing value for boolean URI options in my mind) but I think the most consistent/compatible way of handling such values is to warn-and-continue.

> My reasoning with preferring a conversion to zero/false over ignoring entirely is motivated by thinking that the user has already indicated they intend to use this option and is more likely to just have forgotten to specify a value. 

I agree with this but I do not see how one can state that `false` was the intended value. For example, if a user gives `tls=&tlsInsecure=&retryReads=` did they mean to leave `tls` and `tlsInsecure` at their defaults (`false`) but change `retryReads` from `true` to `false`?

Reply:
> I think it makes more sense to interpret that as leaving all three options at their defaults (i.e. ignoring them in this case).

Fair point. I'd be OK with warning and ignoring in that case. Is there wiggle room to allow some drivers to be more strict and raise errors (for cases where warnings are basically useless)?

Reply:
Agree on this. Especially for options of integer type where the default value is non-zero, it would be safer to use the default value for the option.

Reply:
We should allow drivers that cannot warn to raise errors. How about this wording:
_If the value is the empty string, then a WARN level logging message MUST be issued and the key is ignored. Drivers that cannot log warning messages or where warnings are meaningless MUST raise an error._

Reply:
Replace "are meaningless" with "might be easily overlooked" and that would work for me.

Reply:
I have updated the wording.